src/control/server/instance_exec.go

```diff
@@ -111,18 +111,18 @@ func (srv *IOServerInstance) finishStartup(ctx context.Context, ready *srvpb.Not
 	// number of targets, not number requested when starting
 	srv.setTargetCount(int(ready.GetNtgts()))
 
-	if srv.isMSReplica() {
-		if err := srv.startMgmtSvc(ctx); err != nil {
-			return errors.Wrap(err, "failed to start management service")
-		}
-	}
-
 	if err := srv.loadModules(ctx); err != nil {
 		return errors.Wrap(err, "failed to load I/O server modules")
 	}
 
 	srv.ready.SetTrue()
 
+	for _, fn := range srv.onReady {
```

Comment:
why not move this block into SetTrue() method, then it becomes atomic

Reply:
Not really sure how that would work, since the callbacks return an error. They don't need to be run atomically either.